pypeclub/OpenPype
# Ftrack: Addiotional component metadata

## Brief description
Added ability to add additional metadata into components.

## Description
Additional metadata are not stored under `ftr_meta` (required for play and cinesync) but as keys next to it. It is possible to set which keys are added to component in settings. If value for a key is not available component it is skipped. There are available keys `openpype_version`, `frame_start`, `frame_end`, `duration`, `width`, `height`, `fps` and `codec`.

### Screenshots
![image](https://user-images.githubusercontent.com/43494761/185122014-f8e6e9f5-49c4-4fb1-a036-b92e5af1f7d3.png)

## Testing notes:
1. Add additional metadata keys in `project_settings/ftrack/publish/IntegrateFtrackInstance/additional_metadata_keys` (e.g. `openpype_version`)
2. Publish something which creates components in ftrack
3. Check it's metadata
